### PR TITLE
Improve evolve_as_block speed

### PR DESCRIPTION
**Pull request by TimeEngineer**
In my machine:

# CPU
model name: AMD Ryzen 7 3800X 8-Core Processor
cache size: 512 KB
cpu MHz: 3515.208 (min) - 4397.87 (max)

# Result
Data Size: 524 KB
Before: ~ 10 ms
Now: ~ 0.27 ms

### CODE REVIEW

**Review by FirelightFlagboy — APPROVED** [review]
could be interesting to crate a benchmark in rust https://doc.rust-lang.org/cargo/commands/cargo-bench.html

**Comment by TimeEngineer** [comment]
> could be interesting to crate a benchmark in rust https://doc.rust-lang.org/cargo/commands/cargo-bench.html
Yes it could be great for many operations in Rust

Here the major improvement is binding, we should avoid Vec<u8> in input

**Review by touilleMan — APPROVED** [review]

File: oxidation/crates/parsec_api_crypto/Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ libsodium-sys = { version = "0.2.7", optional = true }
 
 digest = { version = "0.10.1", optional = true }
 generic-array = { version = "0.14.4", optional = true, features = ["serde"] }
-sha2 = { version = "0.10.0", optional = true }
+sha2 = { version = "0.10.0", optional = true, features = ["asm"] }
```
Comment (touilleMan) [inline oxidation/crates/parsec_api_crypto/Cargo.toml:21]: asm feature compatible with wasm32 ?

Comment (TimeEngineer) [reply oxidation/crates/parsec_api_crypto/Cargo.toml:21]: Only x86, x86-64, and (partially) AArch64 architectures are currently supported.

**Pull request merged**